src/pcost.py

```diff
@@ -0,0 +1,32 @@
+import csv
+from pathlib import Path
+from typing import Union
+import logging
+
+
+def portfolio_cost(file_path: Union[str, Path]) -> float:
+    """
+    Calculates the cost of the portfolio by using the amount of shares
+    and the corresponding price.
+
+    :param file_path: Path to the portfolio file
+    :return: The cost of the portfolio
+    """
+    total_cost = 0.0
+    with open(file_path) as f:
+        rows = csv.reader(f)
+        headers = next(rows)
+        for rowno, row in enumerate(rows, 1):
+            if not row:
+                continue
+            rowdict = dict(zip(headers, row))
+            try:
+                shares = int(rowdict["shares"])
+                price = float(rowdict["price"])
+                total_cost += shares * price
+
+            # Exercise asks to catch the error like this
+            except ValueError:
+                logging.info(f"Row {rowno}: Bad row {rowdict}")
```

Comment:
The `logging` module offers different levels:
- debug
- info
- warning
- error
- critical
- exception

The level to choose depends on the "severity" of the problem. Here I think we have 2 options:
- use the warning + log the exception message:
```python
except ValueError as e:
   logging.warning(f"Something went wrong while processing row number {rowno}: {rowdict} - {e}")
```
- use the exception level (the error is displayed automatically):
````python
except ValueError:
    logging.exception(f"Something went wrong while processing row number {rowno}: {rowdict}")
````